vizro-core/src/vizro/actions/_callback_mapping/_callback_mapping_utils.py

```diff
@@ -1,91 +1,30 @@
-        return model.selector.actions
-    elif hasattr(model, "actions"):
-        return model.actions
-    return []
-
-
-def _get_all_actions_chains_on_page(page: Page) -> chain:  # type: ignore[type-arg]
-    """Creates an itertools chain of all ActionsChains present on selected Page."""
-    return chain(*(_get_actions(page_item) for page_item in chain([page], page.components, page.controls)))
-
-
-def _get_model_actions_chains_mapping(page: Page) -> Dict[str, ModelActionsChains]:
-    """Creates a mapping of model ids and ModelActionsChains for selected Page."""
-    model_actions_chains_mapping = {}
-    for page_item in chain([page], page.components, page.controls):
-        model_actions_chains_mapping[page_item.id] = ModelActionsChains(
-            model=page_item, actions_chains=_get_actions(page_item)
-        )
-    return model_actions_chains_mapping
-
-
-def _get_triggered_page(action_id: ModelID) -> Page:  # type: ignore[return]
-    """Gets the page where the provided `action_id` has been triggered."""
-    for _, page in model_manager._items_with_type(Page):
-        if any(
-            action.id == action_id
-            for actions_chain in _get_all_actions_chains_on_page(page)
-            for action in actions_chain.actions
-        ):
-            return page
-
-
-def _get_triggered_model(action_id: ModelID) -> VizroBaseModel:  # type: ignore[return]
-    """Gets the model where the provided `action_id` has been triggered."""
-    for _, page in model_manager._items_with_type(Page):
-        for model_id, model_actions_chains in _get_model_actions_chains_mapping(page).items():
-            if any(
-                action.id == action_id
-                for actions_chain in model_actions_chains.actions_chains
-                for action in actions_chain.actions
-            ):
-                return model_actions_chains.model
-
-
-def _get_components_with_data(action_id: ModelID) -> List[str]:
-    """Gets all components that have a registered dataframe on the page where `action_id` was triggered."""
-    page = _get_triggered_page(action_id=action_id)
-    return [component.id for component in page.components if data_manager._has_registered_data(component.id)]
-
-
 def _get_matching_actions_by_function(page: Page, action_function: Callable[[Any], Dict[str, Any]]) -> List[Action]:
     """Gets list of Actions on triggered page that match the provided action function."""
     return [
         action
-        for actions_chain in _get_all_actions_chains_on_page(page)
+        for actions_chain in page._get_page_actions_chains()
         for action in actions_chain.actions
         if action.function._function == action_function
     ]
 
 
 # CALLBACK STATES --------------
-def _get_inputs_of_controls(action_id: ModelID, control_type: ControlType) -> List[State]:
+def _get_inputs_of_controls(page: Page, control_type: ControlType) -> List[State]:
     """Gets list of States for selected control_type of triggered page."""
```

Comment:
```suggestion
    """Gets list of States for selected `control_type` of triggered `page`."""
```

Reply:
Should this even matter? Happy to follow a guideline here, but so far I don't think we have been consistently adding single quotation marks to model fields and Models in private functions. Shall we start doing that? Or shall we be less strict/pedantic about it? I am fine either way

Reply:
I proposed it here as it helped me read the docstring better/easier in other cases where it was in place. Here e.g. I read the docstring three times to ensure I was getting the meaning (e.g. maybe the triggered page is coming from somewhere else). 

I think we don't need to enforce it, but I think it's always good to make things more readable/improve. In the case below e.g. it was referring even to arguments not present which is not so good.

Given that it's super easy to accept a suggestion it could be a good addition?

Reply:
If it helps you read it, then let's be consistent about it :) 

We can change the relevant parts here in the PR only, but then I'll create a ticket where someone can quickly go through all docstrings and have function arguments in single quotation marks then 👍